pasta/base/token_generator.py

```diff
@@ -326,6 +328,55 @@ def predicate(token):
       self._loc = tok.end
     return content
 
+  def fstr(self):
+    """Parses an fstring, including subexpressions.
+
+    Returns:
+      A generator function which, when repeatedly reads a chunk of the fstring
+      up until the next subexpression and yields that chunk, plus a new token
+      generator to use to parse the subexpression. The subexpressions in the
+      original fstring data are replaced by placeholders to make it possible to
+      fill them in with new values, if desired.
+    """
+    def fstr_parser():
+      # Reads the whole fstring as a string, then parses it char by char
+      str_content = self.str()
+      chars = enumerate(str_content)
```

Comment:
would call this like chars_and_idxs or enumerated_chars or something so i know its not a flat list